org.lflang/src/org/lflang/generator/c/CGenerator.xtend

```diff
@@ -4294,8 +4293,9 @@ class CGenerator extends GeneratorBase {
                 reactionInstance.definition
             )) {
                 val reactionStructName = '''«selfStructName(reactionInstance.parent)»->___reaction_«reactionCount»'''
-                val reactionIndex = "0x" + (reactionInstance.deadline.toNanoSeconds.shiftLeft(16)).or(
-                    new BigInteger(reactionInstance.level.toString)).toString(16) + "LL"
+                // xtend doesn't support bitwise operators...
+                val indexValue = CGeneratorExtension.longOr(reactionInstance.deadline.toNanoSeconds << 16, reactionInstance.level)
```

Comment:
`longOr` is unrelated to federated execution.

Reply:
Yeah that's true... but we currently don't have a utility class for this kind of things... Maybe this should go into a   `MissingLanguageFeaturesUtil`... jk I'll find something